### PR TITLE
Split up cookie headers that are added via HttpListenerResponse.WebHeaders.Add

### PR DESCRIPTION
**Pull request by rmkerr**
This fixes an inconsistency between the Windows and Managed implementation of  HttpListenerResponse.

Basically, there are two ways to add cookies to an HttpListenerResponse. The 'usual' way is to add them to the cookie collection:
```
response.Cookies = cookies;
```
On Unix and on Windows that yields the following Set-Cookie header:
```
Set-Cookie: name1=value1, name2=value2
```
In this case though the user is setting the headers by adding them directly:
```
foreach (var c in cookies)
{
    HttpListenerResponse.Headers.Add("Set-Cookie",c);
}
```
According to the relevant documentation for WebHeaderCollection.Add headers added more than once will be added in a comma separated list. So, even adding the headers directly, we expect the same result. That is true on Unix, and we get the following header:
```
Set-Cookie: name1=value1,name2=value2
```
However, on Windows we see an unexpected result:
```
Set-Cookie: name1=value1
Set-Cookie: name2=value2
```
While both behaviors are acceptable, I think that the Windows behavior was intentionally added. That is based off of this comment in the relevant code:
https://github.com/dotnet/corefx/blob/3db29b721a45fb4f300e27c5df5750f57b38ec45/src/System.Net.HttpListener/src/System/Net/Windows/HttpListenerResponse.Windows.cs#L460-L482
This PR ensures that the managed implementation follows the same behavior as the Windows version. I think that should also update the documentation to reflect this behavior, since it seems to be by design.

Fixes: #22910

### CODE REVIEW

**Comment by rmkerr** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

File: src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs [hunk 1]
```diff
@@ -170,6 +170,24 @@ public static IEnumerable<object[]> Cookies_TestData()
...
+        {
+            HttpListenerResponse response = await GetResponse();
+            response.Headers.Add("Set-Cookie", "name1=value1");
+            response.Headers.Add("Set-Cookie", "name2=value2");
+            response.Headers.Add("Set-Cookie", "name3=value3");
+            response.Headers.Add("Set-Cookie", "name4=value4");
+
+            response.Close();
+
+            string clientResponse = GetClientResponse(224);
```
Comment (davidsh) [inline src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs]: What is "224"?  Perhaps using named parameters in the call site here or adding a comment would help.

Comment (rmkerr) [reply src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs]: It's the size of the expected message, and is expected by this somewhat unwieldy helper method:
https://github.com/dotnet/corefx/blob/fb0cbae3e0275fdf9b2b7bc734422c88285acafb/src/System.Net.HttpListener/tests/HttpListenerResponseTests.cs#L34-L50
Using a named parameter would definitely make things clearer, so I'll add that,

File: src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs [hunk 2]
```diff
@@ -166,10 +166,28 @@ public static IEnumerable<object[]> Cookies_TestData()
...
+            HttpListenerResponse response = await GetResponse();
+            response.Headers.Add("Set-Cookie", "name1=value1");
+            response.Headers.Add("Set-Cookie", "name2=value2");
+            response.Headers.Add("Set-Cookie", "name3=value3");
+            response.Headers.Add("Set-Cookie", "name4=value4");
+
+            response.Close();
+
+            string clientResponse = GetClientResponse(expectedLength:224);
+            Assert.Contains($"\r\nSet-Cookie: name1=value1\r\n", clientResponse);
```
Comment (davidsh) [inline src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs]: nit: using '$' interpolation isn't required here as there are no variables to evaluate in these strings.

Comment (rmkerr) [reply src/System.Net.HttpListener/tests/HttpListenerResponseTests.Cookies.cs]: Good catch. Fixed throughout the whole file.

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs [hunk 3]
```diff
@@ -315,7 +315,15 @@ private static string FormatHeaders(WebHeaderCollection headers)
                     {
                         if (anyValues)
                         {
-                            sb.Append(", ");
+                            if (key.Equals(HttpKnownHeaderNames.SetCookie, StringComparison.InvariantCultureIgnoreCase) ||
+                                key.Equals(HttpKnownHeaderNames.SetCookie2, StringComparison.InvariantCultureIgnoreCase))
```
Comment (davidsh) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs]: It's possible that `StringComparison.OrdinalIgnoreCase` would be better (and faster) here.

@stephentoub PTAL

Comment (stephentoub) [reply src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs]: Yeah, let's use ordinal ignore case.

**Comment by rmkerr** [comment]
The only failure here is in System.Data.SqlTypes and doesn't touch any of the relevant code, so I'm going to go ahead and merge this.

**Pull request merged**